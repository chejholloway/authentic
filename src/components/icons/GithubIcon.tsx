import React from "react";
import github from "../../assets/images/github-full-svgrepo-com.svg";
/**
 * Renders the GitHub icon as an image.
 *
 * This component displays the GitHub logo as an image. You can provide an optional
 * CSS class name to style the icon as needed.
 *
 * @component
 * @example
 * ```tsx
 * <GithubIcon className="text-gray-600" />
 * ```
 * @defaultprops
 * @param {object} props - The component props.
 * @param {string} [props.className] - An optional CSS class name to apply custom styling.
 */
interface GithubIconProps {
  className?: string;
}


const GithubIcon: React.FC<GithubIconProps> = ({ className }) => {
  return (
    <img
      className={`h-10 object-cover ${className}`}
      src={github}
      alt="Github Text Logo"
    />
  );
};

export default GithubIcon;
