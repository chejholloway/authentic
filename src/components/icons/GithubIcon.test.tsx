import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer"; // Import react-test-renderer
import GithubIcon from "./GithubIcon";

describe("GithubIcon", () => {
  test("renders Github icon", () => {
    render(<GithubIcon className="test-class" />);

    // Assert that the Github icon image is rendered with the provided class
    const githubIcon = screen.getByAltText("Github Text Logo");
    expect(githubIcon).toBeInTheDocument();
    expect(githubIcon).toHaveClass("h-10 object-cover test-class");
  });

  test("matches snapshot", () => {
    // Use react-test-renderer to create a snapshot of the component
    const tree = renderer.create(<GithubIcon className="test-class" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
