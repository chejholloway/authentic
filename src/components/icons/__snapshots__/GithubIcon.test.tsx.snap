// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubIcon matches snapshot 1`] = `
<img
  alt="Github Text Logo"
  className="h-10 object-cover test-class"
  src="github-full-svgrepo-com.svg"
/>
`;
